### PR TITLE
docs: agent documentation review pass (LFXV2-2468)

### PR DESCRIPTION
**Pull request by emsearcy**
## Summary

This PR completes a review pass on agent documentation in the lfx-mcp repo (LFXV2-2468), covering three tasks:

### 1. Remove `DEVELOPER.md`

Removed `DEVELOPER.md` after verifying it contained no information missing from `AGENTS.md`. Preserved the flag+env table, stdio OAuth limitation note, and manual stdio/HTTP testing examples in `AGENTS.md` where appropriate.

### 2. Review and refresh `AGENTS.md`

- Replace hardcoded Go and SDK versions with references to `go.mod`
- Replace env-var-only table with a combined flag+env table
- Add stdio OAuth limitation note to Prerequisites
- Remove pedantic/stale sections: Expected JSON-RPC Messages, Future Extensions (already implemented), Debugging Tips (duplicated Logging section, used wrong log pattern)
- Fix SDK reference URL to point to `main` branch
- Fix directory tree to reflect actual `internal/` subdirectories

### 3. Add `.github/copilot-instructions.md` for Copilot cloud agent

Onboards the repository for GitHub Copilot cloud code reviews. Scoped to what a cloud agent needs to generate correct PRs without exploring the repo each time: CI rules (license header, active MegaLinter checks vs. disabled ones), project layout, tool-adding pattern, key conventions, and common pitfalls. Local developer workflow remains in `AGENTS.md`.

### 4. Fixes from Copilot review passes

Addressed 7 review comments, all confirmed accurate against the code:

- `internal/auth/` doc corrected (JWT/API-key verification, not OAuth token exchange — that's in `internal/lfxv2/`)
- Stdio transport note corrected (real limitation is no per-request OAuth context, not tool restriction)
- `defaultTools` pitfall note corrected (controls default enablement, not exclusive activation)
- Env var table: removed nonexistent `-http.public_url` flag, added missing `-committees_as_groups` flag, fixed `-tools` default
- Restored the manual HTTP testing example dropped during the `DEVELOPER.md` consolidation, aligned with the stdio example (both now enable only `hello_world` via `LFXMCP_TOOLS`)
- Removed hardcoded MCP SDK version reference in `copilot-instructions.md`

While verifying the restored HTTP example actually worked end-to-end, found and fixed a real bug in `cmd/lfx-mcp-server/main.go`: koanf's `StringToSliceHookFunc` splits an empty string into `[]string{""}` (length 1), so `len(cfg.MCPAPI.AuthServers) > 0` was always true — OAuth bearer-token verification was unconditionally enabled on the `/mcp` HTTP endpoint even when `-mcp_api.auth_servers` was never configured. Fixed both the flag and env var transform functions to return an empty slice for empty values on comma-separated list fields.

Verified with `make check`, `./scripts/test_server.sh`, and manual stdio/HTTP tests.

### 5. Further review fixes and MegaLinter markdown cleanup

Addressed a second and third round of Copilot review comments:

- Comma-separated list parsing (`-tools`, `-mcp_api.auth_servers`, `-mcp_api.scopes` flags and their env-var equivalents) now trims whitespace via a new `splitTrimmed` helper, preventing malformed values (trailing commas, stray spaces) from producing an empty auth-server URL that would break JWKS verifier initialization. Added a table-driven unit test (`cmd/lfx-mcp-server/main_test.go`) covering this helper.
- Rescoped the license-header-check claim and the YAML line-length claim in `copilot-instructions.md` after verifying both against the actual shared workflow config and `.yamllint` (Markdown isn't checked for license headers; the yamllint `line-length` rule is `warning`-level only and not build-blocking).
- Tagged a bare fenced code block with `text` (MegaLinter `MARKDOWN_MARKDOWNLINT` MD040).

While verifying the MegaLinter markdown checks, found and fixed pre-existing markdown lint issues repo-wide that predate this PR (not part of the original three tasks, but caught while auditing docs):

- Added missing top-level headings to `.agents/skills/upgrade-maintenance/SKILL.md` and `.agents/skills/validate-search-filters/SKILL.md` (MD041)
- Converted bare URLs to angle-bracket links in `validate-search-filters/SKILL.md` (MD034)
- Reformatted markdown tables in `AGENTS.md`, `ARCHITECTURE.md`, `README.md`, and both `SKILL.md` files with `markdown-table-formatter` for consistent column alignment

Verified locally with `markdownlint-cli2` and `markdown-table-formatter --check` against MegaLinter's actual config (`megalinter-reports/IDE-config/.markdownlint.json`), plus `make check` and `go test ./cmd/...`.

### 6. Fail closed on malformed list config; scope package-comment requirement to non-test files

A subsequent Copilot review caught that `splitTrimmed` silently normalized a nonblank-but-malformed value (e.g. `-mcp_api.auth_servers=,`) to an empty slice — indistinguishable from an unset value. Since `runHTTPServer` only enables OAuth bearer-token verification when `len(cfg.MCPAPI.AuthServers) > 0`, this meant a misconfigured (but nonblank) auth-server value would fail open and expose `/mcp` without auth instead of rejecting the bad config.

`splitTrimmed` now returns an error for any nonblank input that produces an empty entry after trimming (lone/leading/trailing/embedded commas, whitespace-only entries); an empty string (`""`, truly unset) still yields an empty slice. Both the flag and env-var parsing paths exit with a clear error message on invalid input. Extended the unit test with malformed-input cases.

Also rescoped the package-doc-comment requirement in `copilot-instructions.md` to non-test `.go` files, verified by running revive directly against individual files (not just `./...`) — the `package-comments` rule never flags `_test.go` files, even checked standalone with no package comment present anywhere else in the package.

A third comment questioning the AGENTS.md claim that Streamable HTTP responses are SSE-framed was investigated and found to be based on an incorrect premise: empirically, `POST /mcp` always returns `text/event-stream` for this server (the SDK actively rejects `Accept: application/json` without `text/event-stream` with a 400), so no doc change was needed there.

Jira: LFXV2-2468

---

🤖 Generated with [GitHub Copilot](https://github.com/features/copilot) (via OpenCode)

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -25,13 +25,18 @@ lfx-mcp/
 ├── cmd/
 │   └── lfx-mcp-server/     # Main application entry point
 ├── internal/
+│   ├── auth/               # OAuth token exchange and JWT validation
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply AGENTS.md]: **AI-assisted**: Confirmed and fixed — internal/auth/ is JWT/API-key verification only; OAuth2 token exchange lives in internal/lfxv2/. Updated the directory tree description in 83b80d1.

File: AGENTS.md [hunk 2]
```diff
@@ -60,17 +65,19 @@ The HTTP server is designed to run across multiple pods without coordination:
...
 ### Prerequisites
 
 ```bash
-# Ensure Go 1.26.0+ is installed
-go version  # Should show go version go1.26.0 or later
+# Verify Go is installed at or above the version in go.mod.
+go version
 ```
 
+> **Note:** The stdio transport only exposes the `hello_world` tool. All LFX data tools require OAuth authentication, which is only supported via the HTTP transport. There is currently no personal access token (PAT) capability in LFX, so running the full server locally for end-to-end use is not practical without a complete OAuth setup.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply AGENTS.md]: **AI-assisted**: Confirmed and fixed — reworded to note stdio's real limitation is lack of per-request OAuth context, rather than implying hello_world is the only registered tool. See 83b80d1.

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+- **JSON schema generation**: The MCP SDK auto-generates schemas from struct tags; no manual schema files.
+- **`schemaCache`**: A package-level cache shared across per-request server instances; do not duplicate it.
+
+---
+
+## Common Pitfalls
+
+- Forgetting the license header on new files is the #1 cause of CI failure.
+- New `.go` files without a package doc comment will fail the revive `package-comments` rule.
+- The `defaultTools` list in `main.go` controls which tools are enabled; adding a Register call without adding the name to `defaultTools` means the tool is never active.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply .github/copilot-instructions.md]: **AI-assisted**: Confirmed and fixed — reworded to clarify defaultTools controls default enablement, and tools can still be activated via -tools/LFXMCP_TOOLS. See 83b80d1.

File: AGENTS.md [hunk 4]
```diff
@@ -489,46 +453,29 @@ The Makefile uses optimized build flags:
...
-./bin/lfx-mcp-server
-```
+The server supports configuration via environment variables with the `LFXMCP_` prefix. **Environment variables override command-line flags.**
+
+| Flag                            | Env Var                               | Default     | Description                                                 |
+|---------------------------------|---------------------------------------|-------------|-------------------------------------------------------------|
+| `-mode`                         | `LFXMCP_MODE`                         | `stdio`     | Transport mode: `stdio` or `http`                           |
+| `-http.host`                    | `LFXMCP_HTTP_HOST`                    | `127.0.0.1` | HTTP server bind address                                    |
+| `-http.port`                    | `LFXMCP_HTTP_PORT`                    | `8080`      | HTTP server port                                            |
+| `-http.public_url`              | `LFXMCP_HTTP_PUBLIC_URL`              | —           | Public URL for HTTP transport (reverse proxies)             |
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply AGENTS.md]: **AI-assisted**: Confirmed and fixed — removed the nonexistent -http.public_url row and added the missing -committees_as_groups flag. See f3c492c.

File: AGENTS.md [hunk 5]
```diff
@@ -489,46 +453,29 @@ The Makefile uses optimized build flags:
...
+| `-debug`                        | `LFXMCP_DEBUG`                        | `false`     | Enable debug logging with source locations                  |
+| `-debug_traffic`                | `LFXMCP_DEBUG_TRAFFIC`                | `false`     | Log outbound LFX API request/response bodies                |
+| `-tools`                        | `LFXMCP_TOOLS`                        | —           | Comma-separated list of tools to enable                     |
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply AGENTS.md]: **AI-assisted**: Confirmed and fixed — corrected the default to defaultTools. See f3c492c.

File: AGENTS.md [hunk 6]
```diff
@@ -60,17 +65,19 @@ The HTTP server is designed to run across multiple pods without coordination:
...
 ### Prerequisites
 
 ```bash
-# Ensure Go 1.26.0+ is installed
-go version  # Should show go version go1.26.0 or later
+# Verify Go is installed at or above the version in go.mod.
+go version
 ```
 
+> **Note:** Stdio mode has no per-request OAuth context, so LFX data tools typically fail auth there. Use HTTP mode with OAuth configured, or enable only `hello_world` via `-tools`/`LFXMCP_TOOLS` for smoke tests. There is currently no personal access token (PAT) capability in LFX, so running the full server locally for end-to-end use is not practical without a complete OAuth setup.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply AGENTS.md:79]: **AI-assisted**: Confirmed and fixed — restored a manual HTTP testing example, aligned with the stdio example (both enable only hello_world via LFXMCP_TOOLS). While verifying it worked, found and fixed a real bug: koanf's StringToSliceHookFunc split empty string values into []string{""} (len 1), so OAuth bearer-token verification was unconditionally enabled on /mcp even with -mcp_api.auth_servers unset. See f3c492c.

File: .github/copilot-instructions.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
+# Copilot Instructions — LFX MCP Server
+
+## What This Repository Does
+
+This is the **LFX MCP Server** — a Model Context Protocol (MCP) implementation in Go that exposes the Linux Foundation's LFX platform as MCP tools for AI agents. It uses the official MCP Go SDK and supports JSON-RPC 2.0 over stdio (dev) and Streamable HTTP (production) transports.
+
+**Key facts**: ~25 Go source files, single binary, no database, stateless HTTP mode. Go module version in `go.mod`, MCP SDK v1.6+, built with `ko` for container images. Deployed to Kubernetes via Helm.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply .github/copilot-instructions.md]: **AI-assisted**: Confirmed and fixed — removed the hardcoded SDK version reference, keeping it version-independent. See f3c492c.

File: AGENTS.md [hunk 8]
```diff
@@ -396,15 +403,29 @@ Focus annotation effort on `ReadOnlyHint` and `DestructiveHint` — those have t
...
 
-Test the server by sending JSON-RPC messages:
+Test the server by sending JSON-RPC messages. `hello_world` is not in `defaultTools`, so enable it explicitly:
 
 ```bash
 # Initialize and call tool
 (echo '{"jsonrpc":"2.0","id":1,"method":"initialize","params":{"protocolVersion":"2024-11-05","capabilities":{},"clientInfo":{"name":"test","version":"1.0.0"}}}';
  echo '{"jsonrpc":"2.0","id":2,"method":"tools/call","params":{"name":"hello_world","arguments":{"name":"Test"}}}';
- sleep 0.5) | ./bin/lfx-mcp-server stdio
+ sleep 0.5) | LFXMCP_TOOLS=hello_world ./bin/lfx-mcp-server stdio
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply AGENTS.md]: Fixed, updated the example to use `-mode=stdio` instead of the invalid positional `stdio` argument, which was stopping Go's flag parser from processing subsequent flags like `-debug`. Good catch!

See commit 56cf7fa.

**Comment by emsearcy** [comment]
## Review Feedback Addressed

Commit: 56cf7fa09285a18d5db05344275d11301a4e75bf

### Changes Made
- **AGENTS.md**: Fixed the stdio manual testing example to use `-mode=stdio` instead of the invalid positional `stdio` argument, which was causing Go's flag parser to stop processing subsequent flags (per copilot-pull-request-reviewer[bot])

### Threads Resolved
1 of 1 unresolved threads addressed in this iteration. All other threads were already resolved from a previous iteration.

File: cmd/lfx-mcp-server/main.go [hunk 9]
```diff
@@ -208,8 +208,14 @@ func main() {
...
+		// Handle comma-separated lists. An empty value must yield an empty slice
+		// rather than falling through as a bare string: mapstructure's
+		// StringToSliceHookFunc splits "" into []string{""} (length 1), which would
+		// make len(cfg.MCPAPI.AuthServers) > 0 true even when unset.
+		if key == "tools" || key == "mcp_api.auth_servers" || key == "mcp_api.scopes" {
+			if value == "" {
+				return key, []string{}
+			}
 			return key, strings.Split(value, ",")
 		}
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply cmd/lfx-mcp-server/main.go:246]: **AI-assisted**: Fixed — added a `splitTrimmed` helper used by both the flag and env-var parsing paths (main.go~180) that trims whitespace and drops empty entries before returning the slice.

File: cmd/lfx-mcp-server/main.go [hunk 10]
```diff
@@ -226,8 +232,12 @@ func main() {
...
-			// Handle comma-separated lists.
-			if (key == "tools" || key == "mcp_api.auth_servers" || key == "mcp_api.scopes") && v != "" {
+			// Handle comma-separated lists. An empty value must yield an empty
+			// slice rather than a bare string (see the flag provider above for why).
+			if key == "tools" || key == "mcp_api.auth_servers" || key == "mcp_api.scopes" {
+				if v == "" {
+					return key, []string{}
+				}
 				return key, strings.Split(v, ",")
 			}
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply cmd/lfx-mcp-server/main.go:270]: **AI-assisted**: Fixed via the same `splitTrimmed` helper applied to the env-var parsing branch.

File: .github/copilot-instructions.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+**Key facts**: ~25 Go source files, single binary, no database, stateless HTTP mode. Go and MCP SDK versions are pinned in `go.mod`. Built with `ko` for container images. Deployed to Kubernetes via Helm.
+
+---
+
+## CI Rules (what breaks the build)
+
+Two workflows run on every PR (`.github/workflows/`):
+
+1. **`license-header-check.yml`** — every tracked source file must begin with the license header.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply .github/copilot-instructions.md]: **AI-assisted**: Reworded to scope the claim to code-type files, since Markdown docs are not checked by license-header-check.yml.

File: cmd/lfx-mcp-server/main.go [hunk 12]
```diff
@@ -171,6 +171,25 @@ var logger *slog.Logger
 // schemas are computed only once rather than on every incoming request.
 var schemaCache = mcp.NewSchemaCache()
 
+// splitTrimmed splits a comma-separated string into a slice, trimming
+// whitespace from each entry and dropping empty entries. An empty input
+// yields an empty (non-nil-length) slice rather than a slice containing a
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply cmd/lfx-mcp-server/main.go]: **AI-assisted**: Reworded to "a non-nil, zero-length slice" — thanks.

File: cmd/lfx-mcp-server/main.go [hunk 13]
```diff
@@ -171,6 +171,25 @@ var logger *slog.Logger
...
+// single empty string.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply cmd/lfx-mcp-server/main.go]: **AI-assisted**: Same fix applied in the follow-up commit — see reply above.

File: .github/copilot-instructions.md [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+```go
+// Copyright The Linux Foundation and contributors.
+// SPDX-License-Identifier: MIT
+```
+
+For YAML, shell, and Makefile use `#` comment syntax. Missing headers are the #1 cause of CI failure.
+
+**Package doc comment** — every `.go` file must have `// Package <name> ...` immediately above the `package` declaration. The `GO_REVIVE` linter enforces the `package-comments` rule.
+
+**YAML formatting** — lines must be ≤120 characters (config: `.yamllint`). Helm templates in `charts/lfx-mcp/templates/` are excluded from YAML linting.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply .github/copilot-instructions.md]: **AI-assisted**: Confirmed — yamllint only fails on `--strict`, which isn't enabled in .mega-linter.yml, so the `line-length` warning rule is advisory only. Reworded accordingly.

File: cmd/lfx-mcp-server/main.go [hunk 15]
```diff
@@ -171,6 +171,25 @@ var logger *slog.Logger
...
+	parts := strings.Split(value, ",")
+	result := make([]string, 0, len(parts))
+	for _, p := range parts {
+		p = strings.TrimSpace(p)
+		if p != "" {
+			result = append(result, p)
+		}
+	}
+	return result
+}
```
Comment (dealako) [inline cmd/lfx-mcp-server/main.go:196]: [minor] Missing unit test for `splitTrimmed`

Issue: `splitTrimmed` fixes a real bug (previously an empty comma-separated config value produced `[]string{""}` instead of `[]string{}`, which could pass `len(...) > 0` checks and feed an empty auth-server URL into OAuth endpoint construction), but has no unit test.
Proof: no `cmd/lfx-mcp-server/main_test.go` exists; the repo has unit-test precedent for pure helpers elsewhere (e.g. `internal/tools/helpers_test.go`, `internal/lfxv2/access_check_test.go`).
Why it matters: this exact class of bug (an empty string surviving as a list entry) is what just broke; a table test locks in the fix and documents the edge cases (empty, whitespace-only, embedded empty entries, leading/trailing commas) for future changes to this function.
Fix: add `cmd/lfx-mcp-server/main_test.go` with a table-driven test over `splitTrimmed`.

File: .github/copilot-instructions.md [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+
+**Other active MegaLinter checks:** `BASH_SHELLCHECK`, `DOCKERFILE_HADOLINT`, `ACTION_ACTIONLINT`, `REPOSITORY_GITLEAKS`, `MARKDOWN_MARKDOWNLINT`.
+
+**Disabled/error-only (will NOT fail CI):** `GO_GOLANGCI_LINT`, `YAML_PRETTIER`, `REPOSITORY_TRIVY`, `REPOSITORY_CHECKOV`, `REPOSITORY_DEVSKIM`, `SPELL_CSPELL`, `SPELL_LYCHEE`, `COPYPASTE_JSCPD`.
+
+---
+
+## Project Layout
+
+```
```
Comment (dealako) [inline .github/copilot-instructions.md]: [nit] Fenced code block missing language tag

Issue: the "Project Layout" tree opens with a bare fence (three backticks, no language) rather than a language tag.
Why it matters: `AGENTS.md` uses a `text`-tagged fence for the equivalent directory tree, and MegaLinter runs `MARKDOWN_MARKDOWNLINT` (listed as an active check in this same file), which includes the fenced-code-language rule.
Fix: tag the fence as `text` to match the sibling tree in `AGENTS.md`.

**Review by dealako — DISMISSED** [review]
Hi @emsearcy — thanks for the thorough pass here. This PR does what it says: removes the redundant `DEVELOPER.md`, tightens up `AGENTS.md` accuracy (versions sourced from `go.mod`, corrected directory tree, corrected stdio/OAuth limitation note, combined flag+env table), and adds a scoped `copilot-instructions.md` for the Copilot cloud agent. The included `splitTrimmed` fix in `main.go` is a genuine, well-reasoned bug fix (empty comma-separated config values no longer produce a spurious one-element `[""]` slice that could otherwise poison auth-server/tool-list config). I also see this PR already went through several rounds of Copilot review with all prior findings addressed and traceable to specific fix commits — nice discipline on that.

I ran a security-focused subagent and a general code-quality subagent in parallel alongside my own read of the diff (including the full `main.go` config-loading section for context) and cross-checked several factual claims in the new docs against the actual code/config (license-header scope, MegaLinter active/disabled linters, yamllint warning-vs-error behavior, `defaultTools` semantics) — all accurate as written.

### Issue Summary

- 🔴 Blocking: 0
- 🟡 Minor: 1 — `splitTrimmed` has no unit test despite fixing a real, previously-shipped config-parsing bug
- ⚪ Nit: 1 — fenced code block in `copilot-instructions.md` missing a language tag (inconsistent with the sibling tree in `AGENTS.md`)
- ❔ Question: 0

No security findings, no merge conflicts.

### Final Decision: ✅ Approved with minor comments

File: cmd/lfx-mcp-server/main.go (hunk 15 above)
Comment (emsearcy) [reply cmd/lfx-mcp-server/main.go:196]: **AI-assisted**: Added `cmd/lfx-mcp-server/main_test.go` with a table-driven test covering empty, whitespace-only, embedded-empty, and leading/trailing-comma cases.

File: .github/copilot-instructions.md (hunk 16 above)
Comment (emsearcy) [reply .github/copilot-instructions.md]: **AI-assisted**: Tagged the fence as `text` to match the AGENTS.md tree. Also ran markdownlint/markdown-table-formatter against MegaLinter's actual config repo-wide and fixed the remaining pre-existing issues found (missing H1s and bare URLs in the two SKILL.md files, table alignment in several docs).

File: AGENTS.md [hunk 17]
```diff
@@ -396,15 +403,29 @@ Focus annotation effort on `ReadOnlyHint` and `DestructiveHint` — those have t
...
+# Start the server, enabling only hello_world.
+LFXMCP_TOOLS=hello_world ./bin/lfx-mcp-server -mode=http &
+
+# Call the tool.
+curl -X POST http://localhost:8080/mcp \
+  -H "Content-Type: application/json" \
+  -d '{"jsonrpc":"2.0","id":1,"method":"tools/call","params":{"name":"hello_world","arguments":{"name":"Test"}}}'
 ```
 
+Responses are returned as Server-Sent Events (SSE) with `event: message` and `data:` fields.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply AGENTS.md:427]: **AI-assisted**: Tested this empirically against a running server. The curl example as written (no `Accept` override, so curl sends its default `Accept: */*`) does get `Content-Type: text/event-stream` back — confirmed SSE, matching the doc. I also tried the exact pattern used in `scripts/test_oauth_config.sh`/`test_oauth_flow.sh` (`curl ... | jq '.'`, no Accept header) and it actually fails with a jq parse error against the raw SSE framing — so those scripts don't currently parse the response as JSON successfully either. Separately, the SDK's StreamableHTTPHandler rejects `Accept: application/json` alone ("Accept must contain both 'application/json' and 'text/event-stream'"), so there's no way to get a bare-JSON response from POST /mcp at all for this server — it's not conditional on negotiation, it's always SSE-framed. Leaving the AGENTS.md wording as-is since it's accurate; the stale test scripts are a pre-existing, separate issue outside this PR's scope.

File: cmd/lfx-mcp-server/main.go [hunk 18]
```diff
@@ -171,6 +171,25 @@ var logger *slog.Logger
...
+func splitTrimmed(value string) []string {
+	if strings.TrimSpace(value) == "" {
+		return []string{}
+	}
+	parts := strings.Split(value, ",")
+	result := make([]string, 0, len(parts))
+	for _, p := range parts {
+		p = strings.TrimSpace(p)
+		if p != "" {
+			result = append(result, p)
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply cmd/lfx-mcp-server/main.go]: **AI-assisted**: Good catch — fixed by rejecting any nonblank input that produces an empty entry after trimming, rather than silently normalizing it to an empty slice. `splitTrimmed` now returns an error for values like `,` or a trailing comma, and both the flag and env-var parsing paths exit with a clear error message instead of failing open. Added table-driven test cases for the malformed inputs.

File: .github/copilot-instructions.md [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+**License header** — always include as the first lines:
+
+```go
+// Copyright The Linux Foundation and contributors.
+// SPDX-License-Identifier: MIT
+```
+
+For YAML, shell, and Makefile use `#` comment syntax. Missing headers are the #1 cause of CI failure.
+
+**Package doc comment** — every `.go` file must have `// Package <name> ...` immediately above the `package` declaration. The `GO_REVIVE` linter enforces the `package-comments` rule.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply .github/copilot-instructions.md]: **AI-assisted**: Confirmed by testing revive directly against individual files (not just `./...`) — the `package-comments` rule never flags `_test.go` files, even when checked standalone with no package comment present anywhere else. Scoped the wording to non-test `.go` files in all three spots.

File: cmd/lfx-mcp-server/main.go [hunk 20]
```diff
@@ -226,9 +258,15 @@ func main() {
...
 			key = strings.ReplaceAll(key, "http_", "http.")
-			// Handle comma-separated lists.
-			if (key == "tools" || key == "mcp_api.auth_servers" || key == "mcp_api.scopes") && v != "" {
-				return key, strings.Split(v, ",")
+			// Handle comma-separated lists. An empty value must yield an empty
+			// slice rather than a bare string (see the flag provider above for why).
+			if key == "tools" || key == "mcp_api.auth_servers" || key == "mcp_api.scopes" {
+				result, err := splitTrimmed(v)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Invalid value for LFXMCP_%s: %v\n", strings.ToUpper(k), err)
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply cmd/lfx-mcp-server/main.go]: **AI-assisted**: Confirmed and fixed — verified empirically with `LFXMCP_TOOLS=, ./bin/lfx-mcp-server`, which was printing `Invalid value for LFXMCP_LFXMCP_TOOLS: ...`. koanf's env.Provider passes the full prefixed name through as `k`, so the extra `LFXMCP_` prepend was wrong. Now prints `Invalid value for LFXMCP_TOOLS: ...`, confirmed against a real run.

File: .github/copilot-instructions.md [hunk 21]
```diff
@@ -0,0 +1,98 @@
...
+// SPDX-License-Identifier: MIT
+```
+
+For YAML, shell, and Makefile use `#` comment syntax. Missing headers are the #1 cause of CI failure.
+
+**Package doc comment** — every non-test `.go` file must have `// Package <name> ...` immediately above the `package` declaration. The `GO_REVIVE` linter enforces the `package-comments` rule. `_test.go` files are exempt.
+
+**YAML formatting** — keep lines ≤120 characters (config: `.yamllint`); this is a warning, not a build failure. Helm templates in `charts/lfx-mcp/templates/` are excluded from YAML linting.
+
+**Other active MegaLinter checks:** `BASH_SHELLCHECK`, `DOCKERFILE_HADOLINT`, `ACTION_ACTIONLINT`, `REPOSITORY_GITLEAKS`, `MARKDOWN_MARKDOWNLINT`.
```
[the remark this answers is not in the document]
Comment (emsearcy) [reply .github/copilot-instructions.md]: **AI-assisted**: Fixed — collapsed the "active"/"disabled" lists into a single terse note that the Go flavor runs its full bundled linter set by default and that `.mega-linter.yml` is the single source of truth for exceptions, rather than maintaining a second, drift-prone list here. See commit 42f6a8e.

**Comment by dealako** [comment]
Hi @emsearcy — thanks for the very thorough follow-up here. I re-read every prior review thread (mine, Copilot's, and the resulting fix commits), re-ran a security-focused and a general code-quality pass independently over the full diff, and manually verified `go build`, `go vet`, `go test`, and `make check` all pass clean.

👏 **Nice work**
- The `splitTrimmed` fix has real teeth now: it not only trims/filters but *rejects* malformed input (`,`, trailing comma, embedded/whitespace-only entries) with a clear error instead of silently normalizing to an empty slice — closing the auth-fail-open risk Copilot flagged in commit `c2440eb`.
- `cmd/lfx-mcp-server/main_test.go` landed with solid table-driven coverage of exactly the edge cases that mattered (empty, whitespace-only, embedded-empty, leading/trailing commas).
- The env-var error-message double-prefixing bug (`LFXMCP_LFXMCP_TOOLS`) was caught and fixed with an empirical repro, not just a guess.
- The MegaLinter "active checks" list in `copilot-instructions.md` was rightly collapsed into a pointer at `.mega-linter.yml` instead of maintaining a second, drift-prone enumeration — directly addressing the completeness concern.
- Nice discipline throughout tying every fix back to a specific commit and, where a suggestion was declined (the SSE-framing wording), backing it with an empirical test against a running server rather than just asserting.

### Revision tracking
Every item from the prior rounds (mine, and Copilot's across several passes) is now ✅ **Resolved**:
- stdio/`hello_world` note, `defaultTools` "never active" wording, `internal/auth/` tree description — fixed in `83b80d1`.
- Nonexistent `-http.public_url` row, missing `-committees_as_groups` flag, missing curl example, hardcoded SDK version — fixed in `f3c492c`.
- Invalid positional `stdio` arg in the example — fixed in `56cf7fa`.
- Comma-list trim/filter bug, "non-nil-length" wording, license-header scope — fixed in `b016cc9`/`97a4800`.
- yamllint warning-vs-failure wording — fixed in `5e57a78`.
- Missing `splitTrimmed` unit test (mine) and missing `text` fence tag (mine) — fixed in `881d4b2`.
- Auth fail-open on malformed separator-only input (Copilot) — fixed in `c2440eb` (now errors instead of failing open).
- `package-comments` scope overstated for test files — fixed and verified against `revive` directly.
- Env-var error double-prefixing — fixed in `0aba157`.
- Incomplete MegaLinter active-check enumeration — fixed in `42f6a8e`.

### New issue scan
No new correctness, security, or test-coverage issues in this diff (`AGENTS.md`, `.github/copilot-instructions.md`, `README.md`, `ARCHITECTURE.md`, both `SKILL.md` files, `main.go`, `main_test.go`).

One out-of-scope note for awareness, not blocking this PR: `resourceMetadataBaseURL, _ := url.Parse(resourceMetadataBase)` in `main.go` ignores a parse error, which could produce a nil-pointer dereference on malformed input — but that line predates this diff, so it's not something this PR introduced. Worth a follow-up issue if it hasn't been tracked already.

### Issue Summary
- 🔴 Blocking: 0
- 🟡 Minor: 0
- ⚪ Nit: 0

No security findings in scope, no merge conflicts, build/vet/test/lint all clean.

### Final Decision: ✅ Approved

**Review by dealako — APPROVED** [review]
Approved — all prior feedback resolved and verified against current HEAD, no new blocking/minor issues, build/vet/test/lint all clean. See summary comment for details.
